README.md

```diff
@@ -10,213 +10,302 @@
 
-```bash
-git clone https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp.git
-```
+<br>
+
+For more info on KVS please see [AWS KVS Documentation](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/what-is-kinesis-video.html) and [Amazon Kinesis Video Streams FAQs](https://aws.amazon.com/kinesis/video-streams/faqs/).
 
-Note: You will also need to install `pkg-config`, `CMake`, `m4` and a build enviroment. If you are building the GStreamer plugin you will also need GStreamer and GStreamer (Development Libraries).
+<br>
 
-Refer to the [FAQ](#FAQ) for platform specific instructions.
+## Quick Start
+### Required Tools
+The following packages are required to build the SDK libraries. Using a package manager such as _Homebrew_ (Mac), _APT_ (Linux), and _Chocolatey_ (Windows) is the prefered method of installation.
+* C++ Compiler (GNU or Clang recommended)
+* `git`
+* `CMake`
+* `pkg-config` for _Mac_/_Linux_, `pkgconfiglite` for _Windows_
+* `m4`
+* _Windows_ only: `nasm` and `strawberryperl`
 
-### Configure
+<br>
 
-Prepare a build directory in the newly checked out repository:
+### Install GStreamer
+If building the samples or the kvssink GStreamer plugin, GStreamer libraries are required.
 
+_Mac_
 ```bash
-mkdir -p amazon-kinesis-video-streams-producer-sdk-cpp/build
-cd amazon-kinesis-video-streams-producer-sdk-cpp/build
+brew install gstreamer gst-plugins-base gst-plugins-good gst-plugins-bad gst-plugins-ugly gst-libav
+```
+_Linux_
+```bash
+sudo apt install libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev gstreamer1.0-plugins-base-apps gstreamer1.0-plugins-bad gstreamer1.0-plugins-good gstreamer1.0-plugins-ugly gstreamer1.0-tools
+```
+_Windows_
+```bat
+choco install gstreamer --version=1.22.8
+choco install gstreamer-devel --version=1.22.8
+```
+#### Verify GStreamer Installation
+Run the following command to display the GStreamer version to confirm the installation was successful:
+```bash
+gst-launch-1.0 --gst-version
 ```
 
-If you are building on Windows you need to generate `NMake Makefiles`, you should run `cmake .. -G "NMake Makefiles"`
+<br>
 
-GStreamer and JNI is NOT built by default, if you wish to build both you MUST execute `cmake .. -DBUILD_GSTREAMER_PLUGIN=ON -DBUILD_JNI=TRUE`
+### Download
+```bash
+git clone https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp.git
```

Comment:
For faster download, let's clone only the master branch.

Reply:
I tested the difference cloning this on my home network. Cloning the full repo, it takes between 1 and 2 seconds to clone and takes up 14 Mb. With `--single-branch` in the command, it still takes between 1 and 2 seconds to clone and takes up 13 Mb.

I don't think these gains are worth the loss of readily being able to checkout remote branches.